src/core/stream/period/get_adaptation_switch_strategy.ts

```diff
@@ -77,6 +79,23 @@ export default function getAdaptationSwitchStrategy(
     return { type: "continue", value: undefined };
   }
 
+  if (adaptation.type === "audio" && audioTrackSwitchingMode === "direct") {
+    const [bufferedMimeType, bufferedCodec] = segmentBuffer.codec?.split(";") ?? [];
+    const codec = bufferedCodec?.split("\"")[1];
+    const representations = adaptation.getPlayableRepresentations();
+
+    for (let i = 0, len = representations.length; i < len; i += 1) {
+      const representation = representations[i];
+      if (bufferedMimeType !== representation.mimeType) {
+        return { type: "needs-reload", value: undefined };
+      }
+
+      if (codec?.split(".")[0] !== representation.codec?.split(".")[0]) {
+        return { type: "needs-reload", value: undefined };
+      }
+    }
+  }
```

Comment:
We want to always reload when we switching to another AdaptationSet now. Here it is still only done based on the codec.
Also check that we're currently playing that Period to avoid reloading when the next Period is pre-loaded.

I would do the same thing than what we do for the video track for now.